app/main.py

```diff
@@ -1,3 +1,38 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+import os.path
+
+from app.customer import Customer
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+    current_directory = os.path.dirname(os.path.abspath(__file__))
+    file_path = os.path.join(current_directory, "config.json")
+    with open(file_path, "r") as file:
+        config = json.load(file)
+        fuel_price = config.get("FUEL_PRICE")
+        for customer_ in config.get("customers"):
+            customer = Customer(customer_)
+            print(f"{customer.name} has {customer.money} dollars")
+            lowest_price = 10000000
+            optimal_place = ""
+            for shop_ in config.get("shops"):
+                shop = Shop(shop_)
+                cost_of_trip = shop.calculate_trip_price(customer, fuel_price)
+                print(f"{customer.name}'s trip to the {shop.name} "
+                      f"costs {cost_of_trip}")
+                if cost_of_trip < lowest_price:
```

Comment:
What if lowest cost of trip is  20000000?) You don't need hardcode `lowest_price` as  10000000. It is better create separate function for this logic. And you should operate only data that you have, don`t add additional numbers)